scss/components/_off-canvas.scss

```diff
@@ -266,16 +340,35 @@ $content: $maincontent-class
   transform: none;
   z-index: $zindex;
   transition: none;
+  visibility: visible;
 
   @if not $offcanvas-fixed-reveal {
     position: absolute;
   }
 
+  @at-root .#{$content}.has-reveal-#{$position} {
+    margin-#{$position}: $offcanvas-size;
+  }
+
+  // backwards compatibility (prior to v6.4)
   & ~ .#{$content} {
     margin-#{$position}: $offcanvas-size;
   }
 }
 
+/// Overrides the off-canvas styles
+@mixin in-canvas() {
```

Comment:
Is there a need for `!important` here? Generally we try and avoid the use of this whenever possible.